kernel/init.c

```diff
@@ -311,24 +311,19 @@ static int do_device_init(const struct device *dev)
 {
 	int rc = 0;
 
+	/* device initialization has been invoked */
+	dev->state->initialized = true;
```

Comment:
Seems questionable to me to flag this **before** making the init call.  Are you sure this is what the docs require?  Maybe the docs are just wrong?